src/logs/metadata_engine/metadata_engine.py

```diff
@@ -127,8 +130,8 @@ def apply(self, record: Dict, parsed_record: Dict):
             # No matching rule has been found, applying the default rule
             if self.default_rule:
                 _apply_rule(self.default_rule, record, parsed_record)
-        except Exception:
-            logging.exception(f"Encountered exception when running Rule Engine")
+        except Exception as ex:
+            logging.exception(f"Encountered exception when running Rule Engine. ")
```

Comment:
Same, ex can be logged? If not logged than "as ex" not needed. I don't know the reason why metadata engine does not log exceptions but there might be some. Theoretically there could be lots of exceptions, one exception for one log could kill your cloudwatch bill. Maybe some logging with throttling would be nice. I don't think you should do it here though

Reply:
Yeah, let's not implement throttling here. This PR is huge anyway :)